process/interceptors/factory/interceptedTxDataFactory.go

```diff
@@ -78,31 +86,37 @@ func NewInterceptedTxDataFactory(argument *ArgInterceptedDataFactory) (*intercep
 	if check.IfNil(argument.TxSignHasher) {
 		return nil, process.ErrNilHasher
 	}
+	if check.IfNil(argument.EpochNotifier) {
+		return nil, process.ErrNilEpochNotifier
+	}
 
 	interceptedTxDataFactory := &interceptedTxDataFactory{
-		protoMarshalizer:          argument.ProtoMarshalizer,
-		signMarshalizer:           argument.TxSignMarshalizer,
-		hasher:                    argument.Hasher,
-		keyGen:                    argument.KeyGen,
-		singleSigner:              argument.Signer,
-		pubkeyConverter:           argument.AddressPubkeyConv,
-		shardCoordinator:          argument.ShardCoordinator,
-		feeHandler:                argument.FeeHandler,
-		whiteListerVerifiedTxs:    argument.WhiteListerVerifiedTxs,
-		argsParser:                argument.ArgsParser,
-		chainID:                   argument.ChainID,
-		minTransactionVersion:     argument.MinTransactionVersion,
-		epochStartTrigger:         argument.EpochStartTrigger,
-		enableSignTxWithHashEpoch: argument.EnableSignTxWithHashEpoch,
-		txSignHasher:              argument.TxSignHasher,
+		protoMarshalizer:            argument.ProtoMarshalizer,
+		signMarshalizer:             argument.TxSignMarshalizer,
+		hasher:                      argument.Hasher,
+		keyGen:                      argument.KeyGen,
+		singleSigner:                argument.Signer,
+		pubkeyConverter:             argument.AddressPubkeyConv,
+		shardCoordinator:            argument.ShardCoordinator,
+		feeHandler:                  argument.FeeHandler,
+		whiteListerVerifiedTxs:      argument.WhiteListerVerifiedTxs,
+		argsParser:                  argument.ArgsParser,
+		chainID:                     argument.ChainID,
+		minTransactionVersion:       argument.MinTransactionVersion,
+		epochStartTrigger:           argument.EpochStartTrigger,
+		enableSignedTxWithHashEpoch: argument.EnableSignTxWithHashEpoch,
+		txSignHasher:                argument.TxSignHasher,
+		txVersionChecker:            versioning.NewTxVersionChecker(argument.MinTransactionVersion),
 	}
 
+	argument.EpochNotifier.RegisterNotifyHandler(interceptedTxDataFactory)
+
 	return interceptedTxDataFactory, nil
 }
 
 // Create creates instances of InterceptedData by unmarshalling provided buffer
 func (itdf *interceptedTxDataFactory) Create(buff []byte) (process.InterceptedData, error) {
-	enableSignTxWithHash := itdf.epochStartTrigger.Epoch() >= itdf.enableSignTxWithHashEpoch
+	enableSignTxWithHash := itdf.epochStartTrigger.Epoch() >= itdf.enableSignedTxWithHashEpoch
```

Comment:
remove this line and use itdf.flagEnableSignedTxWithHash.IsSet() on L136

Reply:
good catch.